src/sentry/static/sentry/app/actionCreators/sentryAppComponents.tsx

```diff
@@ -7,6 +7,11 @@ export async function fetchSentryAppComponents(
   orgSlug: string,
   projectId: string
 ): Promise<SentryAppComponent[]> {
+  // Short-circuit if the API would just 404.
+  if (!projectId) {
```

Comment:
@mgaeta I think it would be better to figure out why `projectId` is missing here since that seems like a bug.